R/arc-read.R

```diff
@@ -2,27 +2,28 @@
 #'
 #' [arc_read()] combines the functionality of [arc_open()] with [arc_select()]
 #' or [arc_raster()] to read an ArcGIS FeatureLayer, Table, or ImageServer to a
-#' `sf` object or object of class `SpatRaster`. `n_max` defaults to 10000 to
-#' avoid unintentionally reading an entire layer with a very large number of
-#' features.
+#' `sf` object or object of class `SpatRaster`. Optionally set, check, or modify
+#' names for the returned data frame or sf object using the `col_names` and
+#' `name_repair` parameters.
 #'
 #' @inheritParams arc_open
-#' @param col_select Columns to select. Alternate argument for specifying fields
-#'   if fields is `NULL`.
-#' @param col_names If `TRUE`, use the default column names for the feature. If
-#'   `col_names` is a character function with the same length as the number of
-#'   columns in the layer, the default names are replaced with the character
-#'   vector. If `col_names` is one less than the length of the default or if
-#'   `col_names` if `FALSE`, the existing sf column name is retained. If
-#'   `col_names` is the string "alias", names are set to match the available
+#' @param col_names Default `NULL`. If `TRUE`, use the default column names for
+#'   the feature. If `col_names` is a character vector with the same length as
+#'   the number of columns in the layer, the default names are replaced with the
+#'   new names. If the length of `col_names` is one less than the length of the
+#'   default column names, the existing sf column name is retained. If
+#'   `col_names` is the string `"alias"`, names are set to match the available
 #'   alias names for the layer.
-#' @param n_max Maximum number of records to return. Defaults to 10000 or an
-#'   option set with `options("arcgislayers.n_max" = <max records>)`
+#' @param col_select,fields Default `NULL`. A character vector of the field
+#'   names to be returned. By default, all fields are returned. `fields` is
```

Comment:
No - they have the same definition – `col_select` takes precedence and is offerred for consistency with the readr style functions. I can split it back up if you prefer.

Reply:
I just went ahead and split it up.

Reply:
Just saw your other comment about dropping the fields parameter.

My only concern about removing the fields parameter is that if a user supplies the `fields` parameter *and* `col_select`, it will error instead of just ignoring `fields` and the error may be confusing or difficult to trouble-shoot. If you want to remove it, I think it may be a good idea to add a `call` parameter to `arc_select()` so the error message is properly assigned to `arc_read()` instead of appearing to be an error with `arc_select()`.